src/MongoDB.Driver.Core/Core/Operations/AsyncCursor.cs

```diff
@@ -120,6 +148,22 @@ public IEnumerable<TDocument> Current
             }
         }
 
+        /// <inheritdoc/>
+        public bool IsEmpty
+        {
+            get
+            {
+                var batch = _firstBatch ?? _currentBatch;
```

Comment:
The spec line: 
`The initial aggregate response did not include a documentResumeToken (indicating no results).`
I guess having `isEmpty` flag is easier than additional document parsing, since all required varaibles to calculate batch size is already presented.

Reply:
As written this property has odd semantics. If called before MoveNext is ever called it returns whether the first batch is empty. If called after MoveNext is called it returns whether the *current* batch is empty. So initially it is looking forward and subsequently it is looking backwards.

Since we only need this to tell if the first batch was empty let's rename and simplify to:

    public bool WasFirstBatchEmpty => _wasFirstBatchEmpty;

And initialize _wasFirstBatchEmpty in the constructor:

     _wasFirstBatchEmpty = firstBatch.Count == 0;

Reply:
Done